docs/dev/pagination-v2-follow-up_a.md

```diff
@@ -0,0 +1,81 @@
+# Pagination in v2 SQL Engine
+
+Implementing pagination in the OpenSearch SQL plugin can be partitioned along several dimensions:
+1. SQL query type,
+1. Size of queried dataset,
+1. Number of open cursors,
+1. SQL node load balancing and failover.
+
+## SQL Query Type
+The two main query types are filter-and-project and aggregation queries. 
+
+V1 engine only supports cursors for filter-and-project queries.
+
+Therefore, discussion of pagination in V2 engine will also be limited to filter-and-project queries.
+
+Pagination of aggregation queries can be considered after V1 engine is deprecated.
+
+## Size of Queried Dataset
+
+OpenSearch provides several data retrival APIs that are optimized for different use cases.
+
+At this time, SQL plugin concerns only uses simple search API and scroll API.
```

Comment:
What are the tradeoffs between using:
* scroll API (https://opensearch.org/docs/latest/api-reference/scroll/), vs
* from/size (https://opensearch.org/docs/latest/opensearch/search/paginate/#the-from-and-size-parameters) vs
* search_after (https://opensearch.org/docs/latest/opensearch/search/paginate/#the-search_after-parameter), vs
* point-in-time (https://opensearch.org/docs/latest/opensearch/point-in-time)

Since all of these are available, we need to argue for the best experience for the user and the best performance for our system.  We should probably detail the pro/cons for each offer.

Reply:
Also, we should include links to the scroll API (and anything else referenced in this document).

Reply:
However V2 engine gets data from OpenSearch, we need to be able to paginate it. 

V2 engine encapsulates retrieval strategy it chose from the execution engine. Currently it picks between all-at-once or scroll requests but it looks the same by the time we send it to the client.